src/service/cid-impl.js

```diff
@@ -181,7 +181,12 @@ export class Cid {
           ViewerCidApi.scopeOptedInForCidApi(this.ampdoc.win, scope);
       if (apiClient) {
         return this.cidApi_.getScopedCid(
-            apiClient, scope, getCidStruct.cookieName);
+            apiClient, scope, getCidStruct.cookieName).then(scopedCid => {
+              if (scopedCid) {
+                return (scopedCid == OPT_OUT) ? null : scopedCid;
+              }
+              return getOrCreateCookie(this, getCidStruct, persistenceConsent);
```

Comment:
You could now delete the `getCookie` calls (e.g. line 101) for fallback in cid-api, right?